anta/cli/get/utils.py

```diff
@@ -47,3 +47,37 @@ def create_inventory(inv: List[Dict[str, Any]], directory: str, container: str)
     with open(out_file, "w", encoding="UTF-8") as out_fd:
         out_fd.write(yaml.dump(i))
     logger.info(f"Inventory file has been created in {out_file}")
+
+
+def create_inventory_from_ansible(inventory: str, directory: str, ansible_root: str = None, output_file: str = 'inventory.yml') -> None:
+    def deep_yaml_parsing(data: Dict[str, Any], hosts: Union[None, List[Dict[str, str]]] = None) -> Union[None, List[Dict[str, str]]]:
+        """Deep parsing of YAML file to extract hosts and associated IPs"""
+        if hosts is None:
+            hosts = []
+        for key, value in data.items():
+            if isinstance(value, dict) and "ansible_host" in value.keys():
+                hosts.append({"name": key, "host": value["ansible_host"]})
+            elif isinstance(value, dict):
+                deep_yaml_parsing(value, hosts)
+            else:
+                # print(f"Key: {key}\n  Value: {value}")
+                return hosts
+        return hosts
+
+    i: Dict[str, Dict[str, Any]] = {AntaInventory.INVENTORY_ROOT_KEY: {"hosts": []}}
+    ansible_inventory = yaml.safe_load(open(inventory))
+    if ansible_root not in ansible_inventory.keys():
+        logger.error(f"Group {ansible_root} not in ansible inventory {inventory}")
+        raise (f"Group {ansible_root} not in ansible inventory {inventory}")
+    if ansible_root is None:
+        ansible_hosts = deep_yaml_parsing(ansible_inventory, hosts=[])
+    else:
+        ansible_hosts = deep_yaml_parsing(ansible_inventory[ansible_root], hosts=[])
+    for dev in ansible_hosts:
+        logger.info(f'   * adding entry for {dev["name"]}')
+        print(f"   * adding entry for {dev}")
+        i[AntaInventory.INVENTORY_ROOT_KEY]["hosts"].append({"host": dev["host"], "name": dev["name"]})
+    out_file = f"{directory}/{output_file}.yml"
+    with open(out_file, "w", encoding="UTF-8") as out_fd:
```

Comment:
You do not need this line. Click opens `outfile_file` for you at the first write.